server/models.py

```diff
@@ -404,7 +405,24 @@ class Batch(BaseModel):
         order_by="BatchResultTallySheet.created_at",
     )
 
-    __table_args__ = (UniqueConstraint("jurisdiction_id", "tabulator", "name"),)
+    # Only one of the following should be populated, enforced via a check constraint defined below
+    last_edited_by_support_user_email = Column(String(200))
+    last_edited_by_jurisdiction_admin_email = Column(String(200))
+    last_edited_by_tally_entry_user_id = Column(
+        String(200), ForeignKey("tally_entry_user.id", ondelete="set null")
+    )
+
+    last_edited_by_tally_entry_user = relationship("TallyEntryUser")
+
+    __table_args__ = (
+        UniqueConstraint("jurisdiction_id", "tabulator", "name"),
+        CheckConstraint(
+            "(cast(last_edited_by_support_user_email is not null as int) +"
+            " cast(last_edited_by_jurisdiction_admin_email is not null as int) +"
+            " cast(last_edited_by_tally_entry_user_id is not null as int)) <= 1",
+            "only_one_of_last_edited_by_fields_is_specified_check",
+        ),
+    )
```

Comment:
Also not totally clear to me is whether I need to tweak the batches query at all:

https://github.com/votingworks/arlo/blob/4fd1051dfefdb2a98cc9d277a189dc8938ab49ae/server/api/batches.py#L100-L112

We're joining with the tally entry users table automatically, and everything is working as far as I can tell. But I'm wondering whether I need to specify any `.options`

Reply:
The way relationships work in SQLAlchemy is that when you access a relationship field on a db object, it fires off a select query for that related object. E.g. accessing `tally_sheet.last_edited_by_tally_entry_user` will trigger a select for `tally_sheet.last_edited_by_tally_entry_user_id` from the `tally_entry_user` table.

In order to join those in during the original query that loads the tally sheets, you do need to specify it with `joinedload`. So in this case, we should have this:

```py
.options(
  joinedload(Batch.result_tally_sheets).joinedload( 
    BatchResultTallySheet.results,
    BatchResultTallySheet.last_edited_by_tally_entry_user
  )
)
```

Reply:
Got it - very helpful! Added `joinedload`s to make sure we use JOINs and not multiple SELECTs (defeating the purpose of this schema switch). (Looked a little bit different than the above since last edited tracking happens at a batch level and not a sheet level.)

https://github.com/votingworks/arlo/pull/1673/commits/7e21986c408a678866ef6712dc80b4f06f021275